src/IlagModule.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2022 New Vector Ltd. t/a Element
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { RuntimeModule } from "@matrix-org/react-sdk-module-api/lib/RuntimeModule";
+import { ModuleApi } from "@matrix-org/react-sdk-module-api/lib/ModuleApi";
+import {
+    JoinFromPreviewListener,
+    RoomPreviewListener,
+    RoomViewLifecycle,
+} from "@matrix-org/react-sdk-module-api/lib/lifecycles/RoomViewLifecycle";
+import { AskNameDialog, AccountModel } from "./components/AskNameDialog";
+import React from "react";
+import { DialogProps } from "@matrix-org/react-sdk-module-api/lib/components/DialogContent";
+
+export default class IlagModule extends RuntimeModule {
+    public constructor(moduleApi: ModuleApi) {
+        super(moduleApi);
+
+        this.moduleApi.registerTranslations({
+            "Creating your account...": {en: "Creating your account..."},
+            "First name": {en: "First name"},
+            "Last name": {en: "Last name"},
+            "Welcome! Enter your name to join": {en: "Welcome! Enter your name to join"},
+        });
```

Comment:
How does this deal with pluralisation?

Reply:
It's just a shortcut to https://github.com/matrix-org/matrix-react-sdk/pull/7886 which is itself a shortcut to the map - people would specify them like the json files and it'd magically work.

Reply:
The JSON files get re-mapped and actually have a different format. I don't think this would work.

The syntax in counterpart is a nested object https://github.com/martinandert/counterpart#pluralization rather than the funky `|one` stuff we invented. `ICustomTranslations` doesn't support the nested object syntax.

`weblateToCounterpart` converts it, but that doesn't appear to be hit in your codepath.

Reply:
then we should fix it independent of this work, I think. It's higher risk for the custom translations than the module work.